packages/lib/src/messaging/Attachments.ts

```diff
@@ -0,0 +1,34 @@
+import { log } from '../shared/log';
+import { Envelop } from './Messaging';
+
+const SUPPORTED_PROTOCOLS = ['http', 'https', 'data'];
+
+function uriCheck(url: URL | undefined): boolean {
+    if (url) {
+        const usesSupportedProtocol =
+            SUPPORTED_PROTOCOLS.find(
+                (protocol) => protocol + ':' === url.protocol,
+            ) !== undefined;
```

Comment:
You can also use the !! operator to evaluate if the value is bullish
` const usesSupportedProtocol :boolean  =
           !! SUPPORTED_PROTOCOLS.find(
                (protocol) => protocol + ':' === url.protocol,
            )`